debug/printf.c

```diff
@@ -34,622 +34,6 @@
 #include <hv_arch.h>
```

Comment:
why dont we move the whole printf.c into lib directly?

Reply:
some functions are used only for debug, so put them under debug/ , and reduce the code size.